### PR TITLE
Convert to sphinx and majorly reorganize

### PR DESCRIPTION
**Pull request by bmbouter**
This introduces a Sphinx docs project and moves almost all of the
content from the README into the sphinx site. After merging, this will
be enabled onto read the docs.

This also reworks the workflows sections significantly and follows the
_snippets approach pioneered in pulp_python.

https://pulp.plan.io/issues/4766
closes #4766

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+# -- Project information -----------------------------------------------------
+
+project = 'Pulp Ansible'
+copyright = '2019, Pulp Ansible Team'
+author = 'Pulp Ansible Team'
+
+# The short X.Y version
+version = '0.1'
+# The full version, including alpha/beta/rc tags
+release = '0.1.0rc5'
```
Comment (daviddavis) [inline docs/conf.py]: I'm not super familiar with how docs work. Is this something we have to remember to bump when we do a release of pulp_ansible? Is there a way to fetch this from setup.py?

File: docs/workflows/mazer-client.rst [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+Install a Collection, hosted by Pulp, using ``mazer``
+=====================================================
+
+
+Mazer Configuration
+-------------------
+
+`Install mazer <https://galaxy.ansible.com/docs/mazer/install.html#latest-stable-release>`_ and
+use the `url` option to point to the `Distribution` your content should fetch from. For example,
+using the `Distribution` created in the sync workflow, the config would be::
```
Comment (daviddavis) [inline docs/workflows/mazer-client.rst]: This paragraph is very vague (ie which url option?)

File: docs/conf.py (hunk 1 above)
Comment (bmbouter) [reply docs/conf.py]: I can add this into my next push. Good idea.

File: docs/workflows/mazer-client.rst (hunk 2 above)
Comment (bmbouter) [reply docs/workflows/mazer-client.rst]: I know :/

Comment (bmbouter) [reply docs/workflows/mazer-client.rst]: These docs need a *lot*. I think we need to fully split roles and collections into their own single large parts of the site.

File: docs/workflows/sync.rst [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+<https://docs.pulpproject.org/en/3.0/nightly/restapi.html#tag/repositories>`_
+
+
+Create a Remote
+---------------
+
+Creating a Remote object informs Pulp about an external content source, which most commonly is
+``https://galaxy.ansible.com/`` or another ``pulp_ansible`` instance. In this case, we will be
+syncing all Roles where ``namespace=elastic`` on Galaxy. You can see those roles
+`here <https://galaxy.ansible.com/api/v1/roles/>`_.
```
Comment (daviddavis) [inline docs/workflows/sync.rst]: Does this need `?namespace=elastic`?

Comment (bmbouter) [reply docs/workflows/sync.rst]: it does, adding in next push.

File: docs/workflows/sync.rst [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+Reference (pulpcore): `Repository API Usage
+<https://docs.pulpproject.org/en/3.0/nightly/restapi.html#tag/repositories>`_
+
+
+Create a Remote
+---------------
+
+Creating a Remote object informs Pulp about an external content source, which most commonly is
+``https://galaxy.ansible.com/`` or another ``pulp_ansible`` instance. In this case, we will be
+syncing all Roles where ``namespace=elastic`` on Galaxy. You can see those roles
```
Comment (daviddavis) [inline docs/workflows/sync.rst]: Here you have 'Roles' and 'roles'.

File: docs/workflows/sync.rst [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    {
+        "_href": "/pulp/api/v3/remotes/ansible/ansible/e1c65074-3a4f-4f06-837e-75a9a90f2c31/",
+    }
+
+.. todo::
+
+    Add a reference link to the live API
+
+
+Sync repository foo with remote
```
Comment (daviddavis) [inline docs/workflows/sync.rst]: Repository and Remote should be titlecased as in the previous sections.

File: docs/workflows/sync.rst [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+-------------------------------
+
+Use the remote object to kick off a synchronize task by specifying the repository to
+sync with. You are telling Pulp to fetch content from the remote and add to the repository.
```
Comment (daviddavis) [inline docs/workflows/sync.rst]: Same here.

File: docs/workflows/sync.rst (hunk 5 above)
Comment (daviddavis) [reply docs/workflows/sync.rst]: Probably should use an article before 'Remote' too.

File: docs/workflows/sync.rst (hunk 4 above)
Comment (daviddavis) [reply docs/workflows/sync.rst]: You should probably do a search for Role in docs and make sure they're titlecased.

File: docs/workflows/upload-role.rst [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+
+Reference (pulpcore): `Artifact API Usage
+<https://docs.pulpproject.org/en/3.0/nightly/restapi.html#tag/artifacts>`_
+
+
+Create a Role from the Artifact
+-------------------------------
+
+Now that Pulp has the tarball, its time to make it into a Role unit of content. We hope to add
```
Comment (daviddavis) [inline docs/workflows/upload-role.rst]: it's time

File: docs/workflows/mazer-client.rst (hunk 2 above)
Comment (daviddavis) [reply docs/workflows/mazer-client.rst]: That makes sense to me. So let's postpone that.

File: docs/conf.py (hunk 1 above)
Comment (bmbouter) [reply docs/conf.py]: I messed with this for a bit, but I want to leave it alone. I think instead we should adopt https://pypi.org/project/bumpversion/ which I know a lot of successful projects rely on. How about that for now?

File: docs/workflows/sync.rst (hunk 4 above)
Comment (bmbouter) [reply docs/workflows/sync.rst]: fixed in next push, ty.

File: docs/workflows/sync.rst (hunk 5 above)
Comment (bmbouter) [reply docs/workflows/sync.rst]: fixed in next push, ty.

File: docs/conf.py (hunk 1 above)
Comment (daviddavis) [reply docs/conf.py]: Sounds good to me. Maybe file an issue if you feel like it.

File: docs/workflows/upload-role.rst (hunk 7 above)
Comment (bmbouter) [reply docs/workflows/upload-role.rst]: agreed, but we need some way to determine the version. We need the tarball to contain that data somehow and that would be above the Role standard, but I think it's what is needed.

File: docs/workflows/sync.rst (hunk 6 above)
Comment (bmbouter) [reply docs/workflows/sync.rst]: fixed in next push.

**Comment by bmbouter** [comment]
@daviddavis I pushed a new version with all those fixes.

File: docs/conf.py (hunk 1 above)
Comment (bmbouter) [reply docs/conf.py]: I made one here  https://pulp.plan.io/issues/4864

**Pull request merged**